src/Testcontainers.LowkeyVault/LowkeyVaultContainer.cs

```diff
@@ -0,0 +1,107 @@
+namespace Testcontainers.LowkeyVault;
+
+/// <inheritdoc cref="DockerContainer" />
+[PublicAPI]
+public sealed class LowkeyVaultContainer : DockerContainer
+{
+    private const string LocalHost = "localhost";
+
+    /// <summary>
+    /// Gets a configured HTTP client
+    /// </summary>
+    private static HttpClient HttpClient => new();
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="LowkeyVaultContainer" /> class.
+    /// </summary>
+    /// <param name="configuration">The container configuration.</param>
+    public LowkeyVaultContainer(LowkeyVaultConfiguration configuration)
+        : base(configuration)
+    {
+    }
+
+    /// <summary>
+    /// Gets the URL of the default vault.
+    /// </summary>
+    /// <returns>The default vault base URL.</returns>
+    public string GetDefaultVaultBaseUrl()
+    {
+        return new UriBuilder(Uri.UriSchemeHttps, Hostname, GetMappedPublicPort(LowkeyVaultBuilder.LowkeyVaultPort)).ToString();
+    }
+
+    /// <summary>
+    /// Gets the full URL of the token endpoint.
+    /// </summary>
+    /// <returns>The full token endpoint URL.</returns>
+    public string GetTokenEndpointUrl()
+    {
+        return new UriBuilder(Uri.UriSchemeHttp, Hostname, GetMappedPublicPort(LowkeyVaultBuilder.LowkeyVaultTokenPort), LowkeyVaultBuilder.TokenEndPointPath).ToString();
+    }
+
+    /// <summary>
+    /// Gets the URL of the vault with a given name.
+    /// <param name="vaultName">the name of the vault.</param>
+    /// </summary>
+    /// <returns>The vault base URL.</returns>
+    public string GetVaultBaseUrl(string vaultName)
+    {
+        // Using `LocalHost` here instead of `Hostname` (which resolves to `127.0.0.1` in this environment) 
+        // to address a compatibility issue with the Java URI parser utilized by the Lowkey Vault client. 
+        // The parser fails to properly handle URIs containing a mix of DNS names and IP addresses, leading to errors. 
+        // For more details, refer to: https://github.com/nagyesta/lowkey-vault/issues/1319#issuecomment-2600214768
+        return new UriBuilder(Uri.UriSchemeHttps, $"{vaultName}.{LocalHost}", GetMappedPublicPort(LowkeyVaultBuilder.LowkeyVaultPort)).ToString();
```

Comment:
This issue is already fixed, correct? Using fixed `localhost` can cause problems in certain environments.

Reply:
so the issue isn't actually fixed as it is a limitation of the way the URI parser in Java works.
However, since that API is a "nice-to-have", I have removed it in the latest commit.